crates/sui-framework/sources/object.move

```diff
@@ -50,10 +50,10 @@ module sui::object {
 
     // === id ===
 
-    /// Create an `ID`. Not to be mistaken for `object::new` which
+    /// Create an `ID` from an unused object address. Not to be mistaken for `object::new` which
     /// generates a new UID.
-    public fun new_id(ctx: &mut TxContext): ID {
-        ID { bytes: tx_context::new_object(ctx) }
+    public fun fresh_id(ctx: &mut TxContext): ID {
```

Comment:
I renamed this mostly to match with `fresh_object_address`

Reply:
I originally tried to get rid of `id_from_bytes` and similar functions since you can now just safely use `address`. But it felt a bit too big of a change, and in particular, it would break the ability for `ID` to be used easily as a key in collections, e.g. `Table`

Reply:
I noticed that you replaced the `ID` in `Referent` with an `address` (which seems valid), but that leads to a question -- is there ever a good reason to use `fresh_id` vs `fresh_object_address` directly?

Reply:
Maybe I should just remove it. It feels a bit strange to have as a part of the framework, and you can always do it yourself by composing with `id_from_address`